### DISCUSSION TITLE
[FR] Optionally show how much each request cost

### DISCUSSION THREAD

**Discussion by NightMachinery** [open-webui/open-webui#2426] [General]
I like to see how much each request I make costs to calibrate my usage better and compare the value to a subscription.

Something like this:
```
Input Tokens: 11
Output Tokens: 663
Total Cost: $0.010 = $0.000 + $0.010, $20 Buys: 2,000.0 (Daily: 66.7)
```

**Comment by djn298h** [comment]
This Function allows you to track for each prompt and sums it up in a cost-$YEAR.json.
I tried it for OpenAI models, but it doesn't seem to calc in the new "reasoning tokens" in the o1-models. 
https://github.com/bgeneto/open-webui-cost-tracker
